M2/Macaulay2/m2/typicalvalues.m2

```diff
@@ -186,9 +193,150 @@ typval = x -> (
 	  else error "typval: expected keyword or function"
 	  )
      else if #x == 5 then typval5f x
-     else error "typval: expected 3, 4, or 5 arguments"
+     else error "typval: expected 3, 4, or 5 arguments";
+     if redefs#?(x#0) then (
+	 f' := x#0;
+	 f := redefs#f';
+	 args := drop(drop(x,-1),1);
+	 installMethod append(prepend(f,args),last x => f');
+	 if args === sequence RR then f Constant := f' @@ numeric
+	 else if #args === 2 then (
+	     if args#0 === RR then f (Constant, args#1) := (x,y) -> f'(numeric x,y);
+	     if args#1 === RR then f (args#0, Constant) := (x,y) -> f'(x,numeric y);
+	     if args === (RR,RR) then f(Constant, Constant) := (x,y) -> f'(numeric x,numeric y); -- phew
+	     ))
      )
-if not member("--no-tvalues", commandLine) then load "tvalues.m2"
+
+load "tvalues.m2"
+
+scan(keys redefs, f -> globalAssign(baseName f,redefs#f));
```

Comment:
You can also do `scanKeys(redefs, (f, m) -> ... )`.

Reply:
or even better, `scanPairs`